### PR TITLE
Fix Minion stats values

### PR DESCRIPTION
**Pull request by LocalIdentity**
We were using the incorrect column in DefaultMonsterStats
The damage difference is ~20% less at level 70 vs the default monster damage table.

I'm not sure if this should apply to Spectre skills so will need to confirm with GGG
edit: applies to all monsters with the IsPlayerMinion tag in MonsterTypes.dat

Damage now lines up with the SRS damage from this spreadsheet https://docs.google.com/spreadsheets/d/1mhh3rQ54_CsRPC37PLqlr_XPIlhy26HP7PjFHSJbUco/edit?usp=sharing

edit 1:
Fixes the handling of the damageFixup stat on minions
Adds support for minons to use the alternate life tables
Add support for handling the baseDamageIgnoresAttackSpeed on some minions

edit2:
Fixes the calculation of Minion armour and evasion base values+
Fix minion accuracy calculations
Use game constants for the damage effectiveness calcs

### CODE REVIEW

File: src/Data/Misc.lua [hunk 1]
```diff
@@ -5,8 +5,11 @@ local data = ...
...
....030000686646, 29.760000228882, 31.579999923706, 33.520000457764, 35.560001373291, 37.720001220703, 40, 42.409999847412, 44.959999084473, 47.639999389648, 50.490001678467, 53.490001678467, 56.659999847412, 60, 63.529998779297, 67.26000213623, 71.199996948242, 75.360000610352, 79.73999786377, 84.370002746582, 89.25, 94.400001525879, 99.839996337891, 105.56999969482, 111.62000274658, 118, 124.73000335693, 131.83000183105, 139.30999755859, 147.19999694824, 155.52000427246, 164.2799987793, 173.5299987793, 183.27000427246, 193.53999328613, 204.36999511719, 215.7799987793, 227.80000305176, 240.46000671387, 253.80999755859, 267.86999511719, 282.69000244141, 298.29000854492, 314.73001098633, 332.04998779297, 350.29000854492, 369.5, 389.73001098633, 411.04000854492, 433.4700012207, 457.08999633789, 481.9700012207, 508.14999389648, 535.71997070312, 564.75, 595.29998779297, 627.46002197266, 661.30999755859, 696.95001220703, 734.45001220703, 773.90997314453, 815.45001220703, 859.15997314453, 905.15002441406, 953.53997802734, 1004.4699707031, 1058.0400390625, 1114.4100341797, 1173.7099609375, 1236.0999755859, 1301.7299804688, 1370.7600097656, 1443.3800048828, 1519.7600097656, 1600.0899658203, }
```
Comment (ifnjeff) [inline src/Data/Misc.lua:12]: For consistency maybe we should name this `monsterAllyDamageTable`, or rename `monsterAllyLifeTable` to `minionLifeTable`

File: src/Modules/CalcActiveSkill.lua [hunk 2]
```diff
@@ -632,9 +632,14 @@ function calcs.buildActiveSkillModList(env, activeSkill)
 			minion.level = m_min(m_max(minion.level,1),100) 
 			minion.itemList = { }
 			minion.uses = activeGrantedEffect.minionUses
-			minion.lifeTable = isSpectre and env.data.monsterLifeTable or env.data.monsterAllyLifeTable
-			local attackTime = minion.minionData.attackTime * (1 - (minion.minionData.damageFixup or 0))
-			local damage = env.data.monsterDamageTable[minion.level] * minion.minionData.damage * attackTime
+			minion.lifeTable = isSpectre and ((minion.minionData.lifeScaling == "AltLife1" and env.data.monsterLifeTable2) or (minion.minionData.lifeScaling == "AltLife2" and env.data.monsterLifeTable3) or env.data.monsterLifeTable) or env.data.monsterAllyLifeTable
+			-- the damage fixup stat applies x% less base Attack Damage and x% more base Attack Speed as confirmed by Openarl Jan 4th 2024
```
Comment (ifnjeff) [inline src/Modules/CalcActiveSkill.lua:636]: For minions that don't have `baseDamageIgnoresAttackSpeed`, a single application of the fixup would accomplish both of these, did you confirm whether it's supposed to double dip for those minions by applying twice?

File: src/Modules/CalcActiveSkill.lua [hunk 3]
```diff
@@ -632,9 +632,14 @@ function calcs.buildActiveSkillModList(env, activeSkill)
...
+			local fixupMult = minion.minionData.damageFixup or 0
+			local attackTime = minion.minionData.attackTime * (1 / 1 + fixupMult)
```
Comment (QuickStick123) [inline src/Modules/CalcActiveSkill.lua:638]:
```suggestion
			local attackTime = minion.minionData.attackTime * (1 / (1 + fixupMult))
```

**Comment by sida-wang** [comment]
Damage fixup implemented in #7268

**Pull request merged**